### DISCUSSION TITLE
Best Way to handle null values in Input Tensor

### DISCUSSION THREAD

**Discussion by rajandeepsingh13** [asus4/tf-lite-unity-sample#301] [Q&A]
**Environment (please complete the following information):**
 - OS/OS Version: Windows
 - Source Version: v2.10
 - Unity Version: Unity 2021.3.5.f1

**Describe the bug**
I am not sure how to pass an Input Tensor that has null values to the Interpreter. My current input is formatted as float?[,,] 3D array but that gives me an "ArgumentException: Object must be an array of primitives." error. What would be the best way to represent an input tensor with null values?

**Error message**
Here is the complete error:
![image](https://user-images.githubusercontent.com/32610322/236903986-1cb2885d-7dc2-4642-9f8b-2f9a61182f99.png)

**Comment by asus4** [comment]
@rajandeepsingh13 nullable is different type. Unbox it before calling SetInputTensorData
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types

**Reply by rajandeepsingh13** [reply]
I see your point, however, looking at nullable arrays, I don't think I can unbox them. They might not be the best data structure to implement what I want to anyway. How would you recommend I implement null value in a non-null array?

For example, I want to pass the following input tensor: [0.23, null, 043, 0.76]. What data structure should/can I use to pass these values to SetInputTensorData? I can't use float[] because I have a null value.

Also, thank you so much for your help so far! You have been awesome!

**Reply by asus4** [reply]
`SetInputTensorData` is a thin wrapper of the TFLite C API. AFAIK, the null value is not supported in the TensorFlow Lite C-API. 

https://github.com/tensorflow/tensorflow/blob/052c468d4bb8c0c90c55c9530d422967baec0ec2/tensorflow/lite/core/c/c_api.h#L459-L462

I think you can set `0` instead of `null`. but if you need to support float?[,,], please open a new issue to the [TensorFlow repo](https://github.com/tensorflow/tensorflow/)
